dashboard/app/models/levels/dsl_defined.rb

```diff
@@ -126,16 +126,16 @@ def self.decrypt_dsl_text_if_necessary(dsl_text)
 
   def clone_with_name(new_name)
     raise "A level named '#{new_name}' already exists" if Level.find_by_name(new_name)
-    level = super(new_name)
     old_dsl = dsl_text
     new_dsl = old_dsl.try(:sub, "name '#{name}'", "name '#{new_name}'")
 
     # raises unless the name is formatted with single, non-curly quotes, e.g.:
     # name 'level-name', or the dsl_text is entirely blank as during unit tests
     raise "name not formatted correctly in dsl text for level: '#{name}'" if old_dsl && old_dsl == new_dsl
 
-    level.update!(dsl_text: new_dsl) if new_dsl
-    level
+    level_params = {}
+    level_params[:encrypted] = encrypted if encrypted
+    self.class.create_from_level_builder({dsl_text: new_dsl}, level_params) if new_dsl
```

Comment:
[re-posting after adding another commit]

 I refactored this to call `create_from_level_builder` directly in order to make this unit-testable. Otherwise, `level = super(new_name)` would return a level with the right value for `encrypted`, but that value would be omitted when `update!` --> `assign_attributes` --> `create_from_level_builder` --> `rewrite_dsl_file` was called, causing the resulting dsl file not to be encrypted.

Reply:
More generally, this closes the gap between what levelbuilders will see on the levelbuilder machine (based on what's in the local db) and what gets output to the filesystem (what will get seeded into the db in other environments).